src/mgmt/tests/mocks.c

```diff
@@ -275,9 +275,32 @@ ds_mgmt_free_pool_list(struct mgmt_list_pools_one **poolsp, uint64_t len)
 int				ds_mgmt_set_prop_return;
 int
 ds_mgmt_pool_set_prop(uuid_t pool_uuid, daos_prop_t *prop,
-					  daos_prop_t **result)
+		      daos_prop_t **result)
 {
-	return ds_mgmt_set_prop_return;
+	daos_prop_t *res_prop;
+	size_t i;
+
+	if (prop == NULL)
+		return -DER_INVAL;
+
+	if (ds_mgmt_set_prop_return != 0)
+		return ds_mgmt_set_prop_return;
+
+	res_prop = daos_prop_alloc(prop->dpp_nr);
+	if (res_prop == NULL)
+		return -DER_NOMEM;
+
+	for (i = 0; i < prop->dpp_nr; i++) {
+		res_prop->dpp_entries[i].dpe_type =
+			prop->dpp_entries[i].dpe_type;
+		res_prop->dpp_entries[i].dpe_val =
+			prop->dpp_entries[i].dpe_val;
+		res_prop->dpp_entries[i].dpe_str =
+			prop->dpp_entries[i].dpe_str;
+	}
+
+	*result = res_prop;
+	return 0;
```

Comment:
A little too much internal logic for the C mocks, in my view. It is nice to have them dumbly return whatever values the test sets up, which lets us test bizarre corner cases. For example, with this mock you can't test the case where the set appears to have succeeded, but the result doesn't include the prop type you set.

Reply:
We're going to have to revisit this set-prop stuff at some point, would prefer to improve the tests then.

Reply:
Sure. I just feel like we should set the mock up a bit more simply to start - have a ds_mgmt_set_prop_result_out that the test sets, and just copy that to the result. You should be able to use daos_prop_dup or daos_prop_copy which simplifies the logic.